showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITUnaryCallable.java

```diff
@@ -0,0 +1,117 @@
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.GrpcStatusCode;
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.api.gax.rpc.CancelledException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.rpc.Status;
+import com.google.showcase.v1beta1.EchoClient;
+import com.google.showcase.v1beta1.EchoRequest;
+import com.google.showcase.v1beta1.EchoResponse;
+import com.google.showcase.v1beta1.EchoSettings;
+import io.grpc.ManagedChannelBuilder;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITUnaryCallable {
+
+  private static EchoClient grpcClient;
+
+  private static EchoClient httpJsonClient;
+
+  @BeforeClass
+  public static void createClients() throws IOException, GeneralSecurityException {
+    // Create gRPC Echo Client
+    EchoSettings grpcEchoSettings =
+        EchoSettings.newBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                InstantiatingGrpcChannelProvider.newBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .build();
+    grpcClient = EchoClient.create(grpcEchoSettings);
+
+    // Create Http JSON Echo Client
+    EchoSettings httpJsonEchoSettings =
+        EchoSettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                EchoSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    httpJsonClient = EchoClient.create(httpJsonEchoSettings);
+  }
+
+  @AfterClass
+  public static void destroyClient() {
+    grpcClient.close();
+    httpJsonClient.close();
+  }
+
+  @Test
+  public void testGrpc_receiveContent() {
+    assertThat(echoGrpc("grpc-echo?")).isEqualTo("grpc-echo?");
+    assertThat(echoGrpc("grpc-echo!")).isEqualTo("grpc-echo!");
+  }
+
+  @Test
+  public void testGrpc_serverResponseError_throwsException() {
+    Status cancelledStatus =
+        Status.newBuilder().setCode(StatusCode.Code.CANCELLED.ordinal()).build();
+    EchoRequest requestWithServerError = EchoRequest.newBuilder().setError(cancelledStatus).build();
+    CancelledException exception =
+        assertThrows(CancelledException.class, () -> grpcClient.echo(requestWithServerError));
+    assertThat(exception.getStatusCode().getCode()).isEqualTo(GrpcStatusCode.Code.CANCELLED);
+  }
+
+  @Test
+  public void testGrpc_shutdown() {
+    assertThat(grpcClient.isShutdown()).isFalse();
+    grpcClient.shutdown();
```

Comment:
QQ, would this make the test dependent on the ordering? i.e. if `testGrpc_shutdown` gets called first and then the client is shutdown, would the other tests still run properly?

Reply:
Great catch! I think you're right, this would make the tests interdependent. Let me look into modifying this.